pallets/native-barrier/src/tests.rs

```diff
@@ -0,0 +1,301 @@
+
+        let daily_limit = 10u128;
+        let future_start = 10 * 86400;
+        let advance_to = 20 * 86400;
+        let future_days = 10;
+        assert_ok!(NativeBarrier::set_daily_xcm_limit(
+            RawOrigin::Root.into(),
+            Some(daily_limit)
+        ));
+        assert_ok!(NativeBarrier::set_start_unix_time(
+            RawOrigin::Root.into(),
+            Some(Duration::from_secs(future_start))
+        ));
+
+        assert_ok!(NativeBarrier::add_accounts_to_native_barrier(
+            RawOrigin::Root.into(),
+            vec![1, 2, 3]
+        ));
+
+        // transfer more than limit should work
+        assert_ok!(Balances::transfer(
+            RuntimeOrigin::signed(1),
+            2,
+            daily_limit + 5
+        ));
+
+        // limit should be multiple of daily limit (now - epoch_start)
+        // roll one day
+        advance_mock_time(Duration::from_secs(advance_to));
+        NativeBarrier::on_initialize(1);
+
+        // check that limit has been updated
+        // transfer more than limit should not work
+        assert_err!(
+            Balances::transfer(
+                RuntimeOrigin::signed(1),
+                2,
+                (future_days as u128 + 1) * daily_limit + 1
+            ),
+            Error::<Runtime>::XcmTransfersLimitExceeded
+        );
+        // transfer under limit should work
+        assert_ok!(Balances::transfer(
+            RuntimeOrigin::signed(1),
+            2,
+            (future_days as u128 + 1) * daily_limit
+        ));
+
+        assert_ok!(NativeBarrier::remove_accounts_from_native_barrier(
+            RawOrigin::Root.into(),
+            vec![1]
+        ));
+
+        // transfer more than limit should work for 1
+        assert_ok!(Balances::transfer(
+            RuntimeOrigin::signed(1),
+            3,
+            (future_days as u128 + 1) * daily_limit + 5
+        ));
+        assert_err!(
+            Balances::transfer(
+                RuntimeOrigin::signed(2),
+                2,
+                (future_days as u128 + 1) * daily_limit + 1
+            ),
+            Error::<Runtime>::XcmTransfersLimitExceeded
+        );
+
+        assert_ok!(NativeBarrier::set_daily_xcm_limit(
+            RawOrigin::Root.into(),
+            None
+        ));
+
+        // transfer more than limit should work for all
+        assert_ok!(Balances::transfer(
+            RuntimeOrigin::signed(1),
+            3,
+            (future_days as u128 + 1) * daily_limit * 2
+        ));
+        assert_ok!(Balances::transfer(
+            RuntimeOrigin::signed(2),
+            3,
+            (future_days as u128 + 1) * daily_limit * 2
+        ));
+        assert_ok!(Balances::transfer(
+            RuntimeOrigin::signed(3),
+            3,
+            (future_days as u128 + 1) * daily_limit * 2
+        ));
+    });
+}
+
+// todo: test that on_inits which don't roll a day will not increase limits.
```

Comment:
todo: test that shows limit doesn't increase on every on-init but only on new days